localstack/services/s3/cors.py

```diff
@@ -0,0 +1,250 @@
+
+        if not is_s3 and not bucket_name:
+            # continue the chain, let the default CORS handler take care of the request
+            return
+
+        # set the service so that the regular CORS enforcer knows it needs to ignore this request
+        context.service = self._service
+
+        request = context.request
+        is_options_request = request.method == "OPTIONS"
+
+        def stop_options_chain():
+            # FIXME: need to add it there as not handled by the serializer, we stop the chain to avoid the request
+            #  being parsed
+            request_id = gen_amzn_requestid_long()
+            response.headers["x-amz-request-id"] = request_id
+            response.headers[
+                "x-amz-id-2"
+            ] = f"MzRISOwyjmnup{request_id}7/JypPGXLh0OVFGcJaaO3KW/hRAqKOpIEEp"
+
+            response.set_response(b"")
+            response.headers.pop("Content-Type", None)
+            chain.stop()
+
+        # check the presence of the Origin header. If not there, it means the request is not concerned about CORS
+        if not (origin := request.headers.get("Origin")):
+            if is_options_request:
+                context.operation = self._get_op_from_request(request)
+                raise BadRequest(
+                    "Insufficient information. Origin request header needed.", HostId=FAKE_HOST_ID
+                )
+            else:
+                # If the header is missing, Amazon S3 doesn't treat the request as a cross-origin request,
+                # and doesn't send CORS response headers in the response.
+                return
+
+        is_origin_allowed_by_default = is_origin_allowed_default(request.headers)
+
+        # The bucket does not exist or does have CORS configured
+        # might apply default LS CORS or raise AWS specific errors
+        if not bucket_name or bucket_name not in self.bucket_cors_index.cors:
+            # if the origin is allowed by localstack per default, adds default LS CORS headers
+            if is_origin_allowed_by_default:
+                add_default_headers(
+                    response_headers=response.headers, request_headers=request.headers
+                )
+                if is_options_request:
+                    stop_options_chain()
+                return
+            # if the origin is not allowed, raise a specific S3 options in case of OPTIONS
+            # if it's a regular request, simply return without adding CORS
+            else:
+                if is_options_request:
+                    if not bucket_name:
+                        message = "CORSResponse: Bucket not found"
+                    else:
+                        message = "CORSResponse: CORS is not enabled for this bucket."
+                    try:
+                        context.operation = self._get_op_from_request(request)
+                    except Exception:
+                        context.operation = context.service.operation_model("GetObject")
```

Comment:
This try catch shouldn't be necessary (as it's already handled in `_get_op_from_request`)

Reply:
Oops, I refactor everything in the `_get_op_from_request` to avoid all these `try...except` but missed one. Thanks! 🙏